src/rdkafka_transport.c

```diff
@@ -498,6 +500,18 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+#if (OPENSSL_VERSION_NUMBER >= 0x0090806fL) && !defined(OPENSSL_NO_TLSEXT)
+	/* If non-numerical hostname, send it for SNI */
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	if ((t = strrchr(name, ':')))
```

Comment:
My understanding is that `nodename` was already a canonicalized version of `origname` and we should always have a port. Isn't that right? If not, yes, an additional field would be helpful (but also, if we have an IPv6, we won't use it as hostname and an IPv6 cannot have a single colon, but I understand, this is not right to write buggy code just because it works).